suite-common/bluetooth/tests/bluetoothReducer.test.ts

```diff
@@ -0,0 +1,243 @@
+            {
+                device: {
+                    id: 'C',
+                    data: [],
+                    name: 'Trezor C',
+                    lastUpdatedTimestamp: 3,
+                },
+                status: null,
+            },
+            // No `B` device present, it was dropped
+            {
+                device: {
+                    id: 'A',
+                    data: [],
+                    name: 'Trezor A',
+                    lastUpdatedTimestamp: 1,
+                },
+                status: { type: 'pairing' }, // Keeps the pairing status
+            },
+        ]);
+    });
+
+    it('changes the status of the given device during pairing process', () => {
+        const nearbyDevice: BluetoothDeviceState<BluetoothDeviceCommon> = {
+            device: {
+                id: 'A',
+                data: [],
+                name: 'Trezor A',
+                lastUpdatedTimestamp: 1,
+            },
+            status: null,
+        };
+
+        const store = configureMockStore({
+            extra: {},
+            reducer: combineReducers({ bluetooth: bluetoothReducer }),
+            preloadedState: {
+                bluetooth: { ...initialState, nearbyDevices: [nearbyDevice] },
+            },
+        });
+
+        store.dispatch(
+            bluetoothConnectDeviceEventAction({
+                id: 'A',
+                connectionStatus: { type: 'pairing', pin: '12345' },
+            }),
+        );
+        expect(store.getState().bluetooth.nearbyDevices).toEqual([
+            {
+                device: nearbyDevice.device,
+                status: { type: 'pairing', pin: '12345' },
+            },
+        ]);
+    });
+
+    it('updates and removes known devices', () => {
+        const store = configureMockStore({
+            extra: {},
+            reducer: combineReducers({ bluetooth: bluetoothReducer }),
+            preloadedState: { bluetooth: initialState },
+        });
+
+        const knownDeviceToAdd: BluetoothDeviceCommon[] = [
+            {
+                id: 'A',
+                data: [],
+                name: 'Trezor A',
+                lastUpdatedTimestamp: 1,
+            },
+            {
+                id: 'B',
+                data: [],
+                name: 'Trezor B',
+                lastUpdatedTimestamp: 2,
+            },
+        ];
+
+        store.dispatch(bluetoothKnownDevicesUpdateAction({ knownDevices: knownDeviceToAdd }));
+        expect(store.getState().bluetooth.knownDevices).toEqual(knownDeviceToAdd);
+
+        store.dispatch(bluetoothRemoveKnownDeviceAction({ id: 'A' }));
+
+        expect(store.getState().bluetooth.knownDevices).toEqual([
+            {
+                id: 'B',
+                data: [],
+                name: 'Trezor B',
+                lastUpdatedTimestamp: 2,
+            },
+        ]);
+    });
+
+    it('removes device from knownDevices when the device is disconnected by TrezorConnect', () => {
```

Comment:
Or what is actually supposed to happen? :thinking: See comment in reducer file
```suggestion
    it('removes device from nearbyDevices when the device is disconnected by TrezorConnect', () => {
```

Reply:
https://github.com/trezor/trezor-suite/pull/17090/commits/1d22b88882b3e493a535f620bcf7eb02d69dcbef